patches/tModLoader/Terraria/ModLoader/GlobalItem.cs

```diff
@@ -194,24 +194,15 @@ public abstract class GlobalItem : ModType
 
 		/// <summary>
 		/// Allows you to temporarily modify this weapon's knockback based on player buffs, etc. This allows you to customize knockback beyond the Player class's limited fields.
-		/// Note that tModLoader follows vanilla principle of only allowing one effective damage class at a time.
-		/// This means that if you want your own custom damage class, all vanilla damage classes must be set to false.
-		/// Vanilla checks classes in this order: melee, ranged, magic, thrown, summon
-		/// So if you set both melee class and another class to true, only the melee knockback will actually be used.
 		/// </summary>
 		/// <param name="item">The item being used</param>
 		/// <param name="player">The player using the item</param>
 		/// <param name="knockback">The knockback</param>
-		public virtual void GetWeaponKnockback(Item item, Player player, ref float knockback) {
+		public virtual void GetWeaponKnockback(Item item, Player player, ref StatModifier knockback, ref float flat) {
```

Comment:
I've made this change, please let me know what you think of it, and update ExampleMod to still compile (I didn't have time)

Reply:
good stuffs